### PR TITLE
refactor: fix build & build:docs

### PR DESCRIPTION
**Pull request by b00ste**
# What does this PR introduce?

Repository cleanup:
- improve `typechain` generation
- improve overloading
- remove unused test

### CODE REVIEW

File: docs/modules/IPFS.md [hunk 1]
```diff
@@ -10,9 +10,9 @@ Returns a valid URL. If it is an IPFS URL (E.g. `ipfs://{hash}`), the IPFS Gatew
 
 #### Parameters
 
-| Name           | Type     | Description                                |
-| :------------- | :------- | :----------------------------------------- |
-| `url`          | `string` | The URL that is to be validated.           |
+| Name | Type | Description |
```
Comment (CJ42) [inline docs/modules/IPFS.md]: What are your prettier settings? It looked better IMO the table before with the spacing between `|`.

File: docs/modules/LSP12.md [hunk 2]
```diff
@@ -25,67 +26,32 @@ v0.0.2
 
 **`Throws`**
 
--   When `newIssuedAssets` is an empty array.
--   When `issuerAddress` is not a valid address.
--   When the contract deployed at `issuerAddress` address does not support the `ERC725Y` interface id.
+- When `newIssuedAssets` is an empty array.
+- When `issuerAddress` is not a valid address.
+- When the contract deployed at `issuerAddress` address does not support the `ERC725Y` interface id.
```
Comment (CJ42) [inline docs/modules/LSP12.md]:
```suggestion
- When the contract deployed at `issuerAddress` does not support the `ERC725Y` interface id.
```

File: src/LSP12/addIssuedAssets/addIssuedAssets.test.ts [hunk 3]
```diff
@@ -40,13 +40,15 @@ describe('addIssuedAssets', () => {
...
             owner.address,
+            0,
             true,
         );
 
         const secondIssuedAsset = await new LSP7Mintable__factory(owner).deploy(
             'SecondTestToken',
             'STT',
             owner.address,
+            0,
```
Comment (CJ42) [inline src/LSP12/addIssuedAssets/addIssuedAssets.test.ts]: Suggest using the constant from lsp-smart-contracts, or adding a comment

```suggestion
            LSP4_TOKEN_TYPES.TOKEN,
```

File: scripts/generateTypechain.bash [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+npx typechain --target=ethers-v6 --out-dir \
+    src/typechain/openzeppelin \
+    ./node_modules/@openzeppelin/contracts/build/contracts/*.json
+
+for FILE in src/typechain/**/*.ts;
+do
+echo '// @ts-nocheck' | cat - $FILE > temp && mv temp $FILE
+done
+
+for FILE in src/typechain/***/**/*.ts;
```
Comment (CJ42) [inline scripts/generateTypechain.bash:18]: What does three stars do? `***`

File: src/LSP2/removeElementFromArrayAndMap/removeElementFromArrayAndMap.ts [hunk 5]
```diff
@@ -1,11 +1,13 @@
 // ethers
 import { BytesLike, concat, isHexString, toBeHex } from 'ethers';
 
-// types
-import { UniversalProfile } from '../../typechain';
+// utils
 import { generateArrayElementKeyAtIndex } from '../generateArrayElementKeyAtIndex';
 import { generateMappingKey } from '../generateMappingKey';
 
+// tyepchain
```
Comment (CJ42) [inline src/LSP2/removeElementFromArrayAndMap/removeElementFromArrayAndMap.ts]:
```suggestion
// typechain
```

File: src/LSP4/getAssetMetadata/getAssetMetadata.test.ts [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
 
 // types
-import { ERC725Y } from '../../typechain';
+import { ERC725Y } from '../../typechain/erc725';
 
 // utils
 import { getAssetMetadata } from '../..';
 
-describe('getDigitalAssetCreators', () => {
+describe.skip('getAssetMetadata', () => {
```
Comment (CJ42) [inline src/LSP4/getAssetMetadata/getAssetMetadata.test.ts:10]: is this intended to be skipped?

Comment (CJ42) [reply src/LSP4/getAssetMetadata/getAssetMetadata.test.ts:10]: Suggest adding a `TODO` comment related to the IPFS issue above.

File: src/LSP4/removeDigitalAssetCreators/removeDigitalAssetCreators.test.ts [hunk 7]
```diff
@@ -39,6 +39,7 @@ describe('removeDigitalAssetCreators', () => {
             'TestToken',
             'TTS',
             owner.address,
+            0,
```
Comment (CJ42) [inline src/LSP4/removeDigitalAssetCreators/removeDigitalAssetCreators.test.ts]: same as above

File: src/helpers/supportsLSPInterface/supportsLSPInterface.ts [hunk 8]
```diff
@@ -1,7 +1,12 @@
 import { BytesLike, Provider, isAddress, isAddressable } from 'ethers';
-import { ERC165, ERC165__factory, InterfaceId, InterfaceIdName } from '../..';
 import { INTERFACE_IDS } from '@lukso/lsp-smart-contracts';
 
+// types
+import { InterfaceId, InterfaceIdName } from '../..';
+
+//typechain
```
Comment (CJ42) [inline src/helpers/supportsLSPInterface/supportsLSPInterface.ts]:
```suggestion
// typechain
```

**Review by CJ42 — COMMENTED** [review]
added review comments

File: scripts/generateTypechain.bash (hunk 4 above)
Comment (b00ste) [reply scripts/generateTypechain.bash:18]: folders, same as 2 stars

Comment (richtera) [reply scripts/generateTypechain.bash:18]: Yea there is no difference between `src/typechain/***/**/*.ts` and `src/typechain/**/*.ts` that I can think of though. Strange.

**Review by richtera — APPROVED** [review]

**Review by CJ42 — APPROVED** [review]

**Pull request merged**